lib/optimizely/tests.js

```diff
@@ -208,11 +209,94 @@ describe('lib/optimizely', function() {
     });
   });
 
+  describe('CustomEventDispatcher', function() {
+    var optlyInstance;
+    var bucketStub;
+    var returnsPromiseEventDispatcher = {
+      dispatchEvent: function(eventObj) {
+        return bluebird.resolve(eventObj);
+      }
+    }
+
+    var createdLogger = logger.createLogger({logLevel: LOG_LEVEL.INFO});
+    var eventDispatcherPromise;
+    beforeEach(function() {
+      bucketStub = sinon.stub(bucketer, 'bucket');
+      eventDispatcherPromise = bluebird.resolve();
+      sinon.stub(eventDispatcher, 'dispatchEvent').returns(bluebird.resolve());
+      sinon.stub(errorHandler, 'handleError');
+      sinon.stub(createdLogger, 'log');
+      sinon.stub(returnsPromiseEventDispatcher, 'dispatchEvent').returns(eventDispatcherPromise);
+    });
+
+    afterEach(function() {
+      bucketer.bucket.restore();
+      eventDispatcher.dispatchEvent.restore();
+      errorHandler.handleError.restore();
+      createdLogger.log.restore();
+      returnsPromiseEventDispatcher.dispatchEvent.restore();
+    });
+
+    it('should execute a custom dispatchEvent\'s promise in activate', function(done) {
+      var instance = new Optimizely({
+        clientEngine: 'node-sdk',
+        datafile: testData.getTestProjectConfig(),
+        errorHandler: errorHandler,
+        eventDispatcher: returnsPromiseEventDispatcher,
+        logger: createdLogger,
+        isValidInstance: true,
+      });
+      bucketStub.returns('111129');
+
+      var activate = instance.activate('testExperiment', 'testUser');;
+
+      assert.strictEqual(activate, 'variation');
+      eventDispatcherPromise.then(function() {
+        var logMessage = createdLogger.log.args[2][1];
+        assert.strictEqual(logMessage, sprintf(LOG_MESSAGES.ACTIVATE_USER,
```

Comment:
Just add a comment here that asserting the log message ensures that we did indeed execute the callback.